src/core/datastore/deltaset.js

```diff
@@ -0,0 +1,99 @@
+import isArray from 'lodash/isArray';
+import { format } from 'url';
+import { Query } from '../query';
+import { KinveyRequest, RequestMethod, AuthType } from '../request';
+import { Client } from '../client';
+import { repositoryProvider } from './repositories';
+import { stripTagFromCollectionName, generateEntityId } from './utils';
+
+const QUERY_CACHE_COLLECTION_NAME = '_QueryCache';
```

Comment:
I feel differently about the naming. Calling it `_QueryCache` in the design was a deliberate decision. I am ok with alternatives that describe what it is - it is a log of queries and corresponding timestamps indicating the last time they were run.  `_QueryHistory`, `_QueryLog`, or equivalent is fine by me.

I want to avoid naming it based on the "delta sync" feature. We could use this table later for other purposes.

Reply:
I hadn't thought of other uses for it. I guess there could be some in the future, I don't know. But "query log" sounds pretty good to me.

I think when the word "cache" is used, it's usually something that has a TTL and what's more important about it is that it is not essential to the work of the system - anything that uses cache, should function without cache, just slower, probably. So I think it would be nice to move away from naming the offline stores "caches" in general. But that's a separate topic :)